### ISSUE TITLE
Formats guidelines

### ISSUE THREAD

**Issue by neerby** [texstremists/Meta#3]
This follows our dicussion on #2 .

In order to foster determinism, clarity and consistency throughout the repos, I suggest adding guidelines concerning main **`README.md`** and **wiki pages**.
In the next comment is a suggestion. Let us edit it as the conversation goes.

**Comment by neerby** [comment]
# Formats



## Main `README.md`

The main `README.md` is the first thing that a user sees when they access a repository. It should be concise and sumarize the main information, that is:

- the repository name;
- a description of what belongs here and indication for contributions;
- an link to the wiki (if existing);
- a list of contents presenting
  - an appropriate hierarchy (list-style or header-style)
  - a linked name to each folder/project
  - a one line description of it
- optional notes.


###### Example


``` markdown
# Repo-name

A short description of what this repository is about ([CONTRIBUTING](<link>)).

Visit the [wiki pages](<link>) for additional contents (documentation, links, etc.).

## Contents

### [folder1](<link>)
Optional description of what can be found here.
- [folder11](<link>): description
  - [file111](<link>): description
  - ...
- [file12](<link>): description
- ...

### [folder2](<link>):
- [file21](<link>):

...

## Notes


```

Here the hierarchy is using headers at first level, which makes more sense if only parent folders are present.

## Wiki pages

One repository's wiki is itself an independant repository that can be cloned, pulled and pushed just the same way. It presents four types of files:

### Home page
The contents of `Home.md`are first to appear; they consist of an introductory sentence, an index and optional notes.

###### Example
```markdown
Welcome to the *Repo-name* wiki! ([CONTRIBUTING](<link>))

# Index
- [page1](<link>): optional short description
- [page2](<link>): optional short description
- ...
```



### Side bar
The contents `_SideBar.md` are always present on the right side of the main page. It contains a short disclaimer mentionning the existence and complementarity of the repository, and a link for contributions.
```markdown
# Disclaimer
This wiki is associtiated with the [`Repo-name` repository](<link>). 
Visit it for related contents (code chunks, sources and complementary files).

([CONTRIBUTING](<link>))
```
### Pages
With format`Page-name.md`, they are the actual content of the wiki. Do not repeat the page name as a header, and add a table of contents (see [how to](https://github.com/texstremists/Meta/wiki/Markdown)) at the beginning.

**Comment by CsatiZoltan** [comment]
What about containing a link to CONTRIBUTION.md?

**Comment by neerby** [comment]
So where shall we drop these guidelines? Code or wiki?

**Comment by CsatiZoltan** [comment]
I am heavily working on the initial version of the contribution guidelines. I think it will be ready by the weekend. I recommend waiting till then and then we can discuss its content on Gitter. The contribution guidelines should go into the main directory of the Meta repo (see its advantages [here](https://help.github.com/articles/setting-guidelines-for-repository-contributors/)).

**Comment by CsatiZoltan** [comment]
The first version is ready. It does not include everything you mentioned - yet. I have the conviction that the classification of rules should not be according to essential-soft, rather code-wiki pages. This is what @neerby  started above, which is clearly for the wiki pages. Since code and wiki editing is quite different, they form two well-distinguishable categories. Let me know what you think.

**Comment by neerby** [comment]
I agree. We could move the first section above (about readme format) to `CONTRIBUTING.md` and turn the document above into a `Contributing.md` wikipage. They might make a few references to eachother.

**Comment by CsatiZoltan** [comment]
So if I understand well:
1. `CONTRIBUTING.md` will contain the format for the main `README.md` needed to be followed for each repository
2. **Meta\wiki\** will have an entry, `Contributing.md` containing the other part of your ideas about wiki creation.

But should really the README.md formatting go into the `CONTRIBUTING.md`? It is done once and for all for each repository and ot is not very likely that new repositories will be created in the future. Shouldn't `CONTRIBUTING.md` focus on what is relevant commit-by-commit?

**Comment by sylph1o** [comment]
I agree with @CsatiZoltan. I suggest to place this information in Meta's wiki; e.g. a page about repository creation, structure or format. Meta's CONTRIBUTING.md may link to it.

Unless I am mistaken, each repository should have a CONTRIBUTING.md pointing to Meta's own. This would give each repository the opportunity to add some information specific to it. E.g. Document-design's CONTRIBUTING.md could specify what is considered 'standard compilation' (see soft rule 4).

**Comment by neerby** [comment]
@CsatiZoltan you're right. I just wish that the README format guideline were explicitely written somewhere: the solution you suggest @sylph1o is totally fine for me.
